mitodl/mit-open
# adding manual migration to fix foreign key type

### What are the relevant tickets?


This contains fixes for #748 and #753

#748 - a bug that can be repro'd in steps below
#753 - migrates certain fields that we need to be nullable for program certificate data sync

### Description (What does it do?)
Full description of bug in #748 is in the ticket.
To repro #748  locally, checkout main. in django shell run:
```
from profiles.models import ProgramLetter, ProgramCertificate
from django.conf import settings
from django.contrib.auth import get_user_model
User = get_user_model()
settings.EXTERNAL_MODELS = []
cert = ProgramCertificate.objects.create(user_email='test@test.com', micromasters_program_id=18, program_title='test')
ProgramLetter.objects.create(certificate=cert, user_id=User.objects.first().id)
```
you should encounter ```invalid input syntax for type integer: "f99df281-93ab-4fc5-9be...```

### How can this be tested?
to test the bug in #748 
 - Checkout this branch. Ensure the migrations have run. re-run the script from above in django shell and it should succeed.

to verify #753:
- checkout this branch - run migrations and ensure that the nullable fields in the programcertificate model line up with the fields in the ticket #753 

### Additional Context


The oddity I saw around #741 was that the script above doesn't appear to fail on main when using django factory models. 
```
from profiles.factories import ProgramLetterFactory, ProgramCertificateFactory
from django.conf import settings
from django.contrib.auth import get_user_model
User = get_user_model()
settings.EXTERNAL_MODELS = []
cert = ProgramCertificateFactory.create(user_email='test@test.com', micromasters_program_id=18, program_title='test')
ProgramLetterFactory.create(certificate=cert, user_id=User.objects.first().id)
```
